src/app/modules/form/form.utils.ts

```diff
@@ -121,3 +121,18 @@ export const getLogicById = (
 
   return form_logics.find((logic) => logicId === String(logic._id)) ?? null
 }
+
+/**
+ * Returns the last form logic in form_logics array
+ * @param form_logics the logics
+ * @returns the last logic if exists, `null` otherwise
+ */
+export const getLastLogic = (
+  form_logics: IFormSchema['form_logics'],
+): ILogicSchema | null => {
+  if (!form_logics) {
+    return null
+  }
+
+  return form_logics[form_logics.length - 1] ?? null
```

Comment:
i think this bugs when form_logics is an empty array. this is because an empty array is a truthy value (`![] === true`) so it wouldn't be caught by the first check.

when it goes to the second check, `[].length === 0`. this means that we are accessing `form_logics[-1]` which evaluates to `undefined` 

i think this could just be avoided using `_.isEmpty` for the initial check and omitting the ternary thereafter (as it's guaranteed to contain >=1 element)

Reply:
Thanks!

To clarify, if I understood your comment correctly
The function returns the last form logic only if it exists in form_logics and null otherwise

The first check is to check that form_logics is not undefined (could be empty array). It returns null if form_logics is undefined.
The second check is to return null if the last form logic is undefined. In this case, if form_logics is empty, form_logics[-1] evaluates to undefined and subsequently to null via the nullish coalescing operator `??`, which is what we want

Reply:
ah ok you're right on the second point, where it defaults to `null` due to the `??` operator. in that case, actually this whole chunk might be better off as `_.isEmpty(form_logics) ? form_logics[form_logics.length - 1] : null`, which would be clearer!

Reply:
Ah but we need the first check if form_logics is defined. This is because we are assuming that the form object returned from the db contains form_logics, but we need to check if this is true i.e. form_logics is defined, and return a DatabaseError otherwise

Reply:
i think for `_.isEmpty()`, it returns `true` as long as collection is empty or it's not a collection type (which is also true for `null | undefined`), which would preserve existing behaviour isn't it?

Reply:
Thanks! You are right and I was about to edit, but then I realised that isEmpty() doesn't type guard for [undefined](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/45521), i.e. typescript will complain that `form_logics[form_logics.length - 1]` is invalid as `form_logics` may be undefined

So I think we will have to stick to the current implementation which explicitly checks that form_logic is defined